runtime/vm/description.c

```diff
@@ -155,16 +157,30 @@ calculateInstanceDescription( J9VMThread *vmThread, J9Class *ramClass, J9Class *
 #endif
 			}
 		}
+
+		ramClass->gcPriorityScanField1 = ramSuperClass->gcPriorityScanField1;
+		ramClass->gcPriorityScanField2 = ramSuperClass->gcPriorityScanField2;
 	}
 
 	/* calculate the description for this class - walk object instance fields and 
 	 * set the corresponding description bit for each 
 	 */
 	{
-		while (walkResult->field) {
+	  	while (walkResult->field) {
 			UDATA slotOffset = walkResult->offset / (objectSlotSize * slotsPerShapeElement);
-#ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
 			U_8 *fieldSigBytes = J9UTF8_DATA(J9ROMFIELDSHAPE_SIGNATURE(walkResult->field));
+			J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(walkResult->field);
+
+			if ((ramClass->gcPriorityScanField1 == 0 || ramClass->gcPriorityScanField2 == 0) && J9UTF8_DATA_EQUALS(J9UTF8_DATA(className), J9UTF8_LENGTH(className), fieldSigBytes + 1, J9UTF8_LENGTH(fieldSig) - 2)) {
```

Comment:
This needs a comment describing why these fields are set and what it's based on.